packages/atomic/src/utils/utils.ts

```diff
@@ -155,3 +155,18 @@ export function getFocusedElement(
 export async function defer() {
   return new Promise<void>((resolve) => setTimeout(resolve));
 }
+
+// https://terodox.tech/how-to-tell-if-an-element-is-in-the-dom-including-the-shadow-dom/
+export function isInDocument(element: Node) {
```

Comment:
In my PR, I'm gonna have to extract `getParent` and `contains` (renamed `isAncestorOf`) from `atomic-focus-trap`. Perhaps you could extract the same methods in your PR and re-use `isAncestorOf`/`contains`.

Alternatively, I'm curious if `closest(element, 'document')` (from utils) would be equivalent to what you're doing.

Reply:
I only moved this method to this file, didn't implement anything! could we look into it in a separate PR? it's so big already 😬